UMLEditor/UMLEditor/Classes/TimeMachine.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace UMLEditor.Classes;
+
+/// <summary>
+/// Class to represent the action history of the UML program
+/// </summary>
+public static class TimeMachine
+{
+    /// <summary>
+    /// Node class to represent a single state in the time machine history.
+    /// </summary>
+    public class Node
+    {
+        /// <summary>
+        /// Reference to previous node
+        /// </summary>
+        public Node? PrevNode;
+        /// <summary>
+        /// Diagram at this node
+        /// </summary>
+        public Diagram? StateDiagram;
+        /// <summary>
+        /// Reference to next node
+        /// </summary>
+        public Node? NextNode;
```

Comment:
It is strange to be rolling your own linked list. Could the time machine be adapted to use a C# standard collection like List?

Reply:
Yeah Robert is right I thought we were going with what Schwartz said and using the C# standard collection.